azure-ts-static-website/staticWebsite.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2016-2018, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+
+export interface StorageStaticWebsiteArgs {
+    accountName: pulumi.Input<string>;
+}
+
+const accountNameProp = "accountName";
+
+// There's currently no way to enable the Static Web Site feature of a storage account via ARM
+// Therefore, we created a custom provider which wraps corresponding Azure CLI commands
+class StorageStaticWebsiteProvider implements pulumi.dynamic.ResourceProvider {
+
+    check = async (olds: any, news: any) => {
```

Comment:
This is a slightly unusual way to write this in TypeScript.  There are two alternatives that might be more natural:
1. Make these normal methods on the class
2. Use an object literal instead of a class (and then just use the single instance of the object literal below instead of `new ...`)

See (e.g.) https://pulumi.io/reference/programming-model/#example-github-labels-rest-api for an example of the latter.